agent-plugins/rfe-dedup/skills/rfe.dedup/scripts/form_groups.py

```
@@ -0,0 +1,468 @@
+        # Restore cut in its neighbors' adjacency lists
+        for nb in adj[cut]:
+            adj[nb].add(cut)
+
+        if len(comps) <= 1:
+            continue
+
+        # cut is an articulation point — assign it to the component with the
+        # most connections to it, breaking ties by average degree.
+        def score(comp):
+            degrees = [
+                all_match_lookup[tuple(sorted([cut, m]))]["match_degree"]
```

Ruff (0.15.12)

[warning] Function definition does not bind loop variable `cut`

(B023)